src/document/json/rga_tree_split.ts

```diff
@@ -586,7 +630,58 @@ export class RGATreeSplit<T extends RGATreeSplitValue> {
       this.treeByIndex.splayNode(node);
     }
 
-    return [changes, createdAtMapByActor];
+    return [changes, createdAtMapByActor, removedNodeMap];
+  }
+
+  public getRemovedNodesLen(): number {
+    return this.removedNodeMap.size;
+  }
+
+  public getCleanupRemovedNodes(ticket: TimeTicket): number {
+    let count = 0;
+    for (const [, node] of this.removedNodeMap) {
+      if (
+        node.getRemovedAt() !== null &&
+        ticket.compare(node.getRemovedAt()) >= 0
+      ) {
+        this.treeByIndex.delete(node);
+        this.purge(node);
+        this.treeByID.remove(node.getID());
+        this.removedNodeMap.delete(node.getID().getAnnotatedString());
+        count++;
+      }
+    }
+
+    return count;
+  }
+
+  // purge physically purge the given node from RGATreeSplit.
```

Comment:
What about using the same comments pattern as other functions?

```suggestion
  /**
   * purge physically purge the given node from RGATreeSplit.
   */
```